### PR TITLE
Display assignment in consumer group member list

### LINKED ISSUE #16: Display partition assignment in consumer group members list
image

Add a new column `Partitions` to indicate which partitions each consumer member holds.

### PR DESCRIPTION
**Pull request by messense**
Fixes #16 

qq20171017-163241 2x

### CODE REVIEW

**Comment by fede1024** [comment]
This is great! However it would be useful to provide the parsing functionality in rdkafka, so that other people in other projects can use it as well. We could add a method to the `GroupMemberInfo` struct, something like:

```rust
enum GroupMemberAssignment {
    Unknown,
    Consumer { topic: String, partitions: Vec<i32>}
}

impl GroupMemberInfo {
[...]
    pub fn assignment(&self) -> GroupMemberAssignment {
        [...]
    }
}
```
or something like that. The member assignment should only be parsed if the protocol type is the correct one.

**Comment by messense** [comment]
I actually try to add the parsing code to `rust-rdkafka` first, but find it nontrivial to do. In order to decode the bytes I need to pull in the byteorder crate and the parsing needs lots of error handling which is not easy in current `rust-rdkafka` setup, namely missing `error-chain` for error handling.

**Comment by fede1024** [comment]
I see. Ok let's leave it here for now then. I'll create an issue in rdkafka and port it at some point (it's not really urgent).

File: src/metadata.rs [hunk 1]
```diff
@@ -158,17 +167,45 @@ pub struct Group {
...
     let mut groups = Vec::new();
     for rd_group in group_list.groups() {
         let members = rd_group.members().iter()
-            .map(|m| GroupMember {
-                id: m.id().to_owned(),
-                client_id: m.client_id().to_owned(),
-                client_host: m.client_host().to_owned()
+            .map(|m| {
+                let mut assigns = vec![];
+                if let Some(assignment) = m.assignment() {
```
Comment (fede1024) [inline src/metadata.rs]: We should have an additional check here to only parse the field if the `protocol_type` is the expected one. Some users might use Kafka's group membership for other kind of groups, and cause the parser to fail or to return unexpected results.

**Pull request merged**